src/ADNLPModels.jl

```diff
@@ -244,22 +244,24 @@ Returns the value `adbackend` from nlp.
 get_adbackend(nlp::ADModel) = nlp.adbackend
 
 """
-    set_adbackend!(nlp, new_adbackend)
-    set_adbackend!(nlp; kwargs...)
+    new_nlp = set_adbackend(nlp, new_adbackend)
+    new_nlp = set_adbackend(nlp; kwargs...)
 
-Replace the current `adbackend` value of nlp by `new_adbackend` or instantiate a new one with `kwargs`, see `ADModelBackend`.
+Create a copy of nlp that replaces the current `adbackend` with `new_adbackend` or instantiate a new one with `kwargs`, see `ADModelBackend`.
 By default, the setter with kwargs will reuse existing backends.
 """
-function set_adbackend!(nlp::ADModel, new_adbackend::ADModelBackend)
-  nlp.adbackend = new_adbackend
-  return nlp
+function _set_adbackend(nlp::ADM, new_adbackend::ADModelBackend) where{ADM}
```

Comment:
```suggestion
function _set_adbackend(nlp::ADM, new_adbackend::ADModelBackend) where{ADM <: AbstractADNLPModel}
```
otherwise nlp.f might lead to an error.